core/src/edu/cornell/gdiac/physics/PlayerController.java

```diff
@@ -161,19 +161,17 @@ public void update(){
             shootCooldown = Math.max(0, shootCooldown - 1);
         }
         if(isInvincible){
-            invincibletimer--;
-            if(invincibletimer <= 0){
-                isInvincible = false;
-                invincibletimer = 50;
+            invincibletimer.update();
+            if(invincibletimer.hasFinished()){
+                invincibletimer.reset();
             }
         }
     }
 
     public void hurt(){
```

Comment:
should it set player to invincible here somehow? It seems like invincible isn't working on this branch at the moment